libxrdp/xrdp_orders_rail.c

```diff
@@ -256,6 +262,8 @@ xrdp_orders_send_window_new_update(struct xrdp_orders *self, int window_id,
     {
         /* titleInfo */
         num_chars = g_mbstowcs(0, window_state->title_info, 0);
+        num_chars = MIN(num_chars, sizeof(twchar) * (g_strlen(window_state->title_info) + 1));
+        num_chars = MAX(num_chars, 0);
```

Comment:
Yes, it happens from g_mbstowcs returning -1.  We can change it to make it more clear what we are checking.  I'm still not sure why "window_state->title_info" get corrupt but this check should be here anyway.